### ISSUE TITLE
[2.0.2] peerDependency warning with the new release

### ISSUE THREAD

**Issue by AviVahl** [chimurai/http-proxy-middleware#703]
### Checks

- [X] I understand project setup issues should be asked on [StackOverflow](https://stackoverflow.com/questions/tagged/http-proxy-middleware) or in [GitHub Discussions](https://github.com/chimurai/http-proxy-middleware/discussions).
- [X] I updated to latest `http-proxy-middleware`.

### Describe the bug (be clear and concise)

Latest version requires `@types/express` as  a peerDependency, printing warnings for non-typescript projects as well.

### Step-by-step reproduction instructions

```shell
mkdir testproxy
cd testproxy
yarn init -y
yarn add http-proxy-middleware

see message:
`warning " > http-proxy-middleware@2.0.2" has unmet peer dependency "@types/express@^4.17.13".`
```

### Expected behavior (be clear and concise)

No warnings when installing package. Don't require users to install `@types` packages as if everyone is using typescript.

### How is http-proxy-middleware used in your project?

```shell
via `webpack-dev-server`, like half the web.
```

### What http-proxy-middleware configuration are you using?

```typescript
Irrelevant
```

### What OS/version and node/version are you seeing the problem?

```shell
Fedora 35; Node 16.13.2; yarn 1.22.17
```

### Additional context (optional)

Regressed by https://github.com/chimurai/http-proxy-middleware/pull/700

**Comment by chimurai** [comment]
Hi thanks for your report.

Those types provide a better developer experience in you IDE with autocompletion; Also in non-typescript projects.
With `@types/express` you'll get autocompletion of express' `req` and `res` objects, which was broken as reported here: https://github.com/chimurai/http-proxy-middleware/issues/692

Perhaps the yarn warning can suppressed by marking as optional:

```json
  "peerDependenciesMeta": {
    "@types/express": {
      "optional": true
    }
  },
```

https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta
>Marking a peer dependency as optional ensures npm will not emit a warning if the _**@types/express**_ package is not installed on the host. This allows you to integrate and interact with a variety of host packages without requiring all of them to be installed.


I have to double check if this will solve the issue.

**Comment by AviVahl** [comment]
Specifying the `peerDependenciesMeta` field would probably solve the warning.

**Comment by chimurai** [comment]
fixed in `v2.0.3`
